File: patch.py
with open(r"C:\Users\jhose\OneDrive\Desktop\PROYECTO DE CODIGO\LEGADO_BIBLICO_PROD\fixed.js", 'r', encoding='utf-8') as f:
    new_lines = f.readlines()

with open(r"C:\Users\jhose\OneDrive\Desktop\PROYECTO DE CODIGO\LEGADO_BIBLICO_PROD\index.html", 'r', encoding='utf-8', errors='replace') as f:
    lines = f.readlines()

# The original array VERSICULOS starts at line 952. Wait, lines in python are 0-indexed.
# In `view_file` the file showed line 951 as `                const VERSICULOS = [`
# So `lines[950]` is where `const VERSICULOS = [` is.
# line 1059 is `                ];` for DEVOCIONALES. So `lines[1058]` is the end.
# Actually, I can just slice the list or construct a new list.

start_idx = -1
end_idx = -1

for i, line in enumerate(lines):
    if "const VERSICULOS = [" in line:
        if start_idx == -1:
            start_idx = i
    if "];" in line and "La Parabola del Trigo y la" in lines[i-1].replace("","") or "La Par" in lines[i-1]:
        # just find where DEVOCIONALES ends
        end_idx = i

# more robust:
for i, line in enumerate(lines):
    if "const VERSICULOS = [" in line:
        start_idx = i
        break

for i in range(start_idx, len(lines)):
    if "];" in line and i > start_idx + 100:  # the devocionales end
        end_idx = i
        break

if start_idx != -1 and end_idx != -1:
    lines = lines[:start_idx] + new_lines + lines[end_idx+1:]
    with open(r"C:\Users\jhose\OneDrive\Desktop\PROYECTO DE CODIGO\LEGADO_BIBLICO_PROD\index.html", 'w', encoding='utf-8') as f:
        f.writelines(lines)
    print(f"Patched strictly from line {start_idx} to {end_idx}.")
else:
    print(f"FAILED to find indices. Start: {start_idx}, End: {end_idx}")

File: _BASURA/PATCHES/patch.js
const fs = require('fs');

const htmlPath = 'C:\\Users\\jhose\\OneDrive\\Desktop\\PROYECTO DE CODIGO\\LEGADO_BIBLICO_PROD\\index.html';
const fixedPath = 'C:\\Users\\jhose\\OneDrive\\Desktop\\PROYECTO DE CODIGO\\LEGADO_BIBLICO_PROD\\fixed.js';

let html = fs.readFileSync(htmlPath, 'utf8'); // It will read invalid chars as \uFFFD
let fixed = fs.readFileSync(fixedPath, 'utf8');

let startIndex = -1;
let endIndex = -1;

const lines = html.split('\n');
for (let i = 0; i < lines.length; i++) {
    if (lines[i].includes('const VERSICULOS = [')) {
        if (startIndex === -1) startIndex = i;
    }
    if (lines[i].includes('];') && i > startIndex + 100 && startIndex !== -1) {
        endIndex = i;
        break;
    }
}

if (startIndex !== -1 && endIndex !== -1) {
    const newLines = lines.slice(0, startIndex).concat(fixed.split('\n'), lines.slice(endIndex + 1));
    fs.writeFileSync(htmlPath, newLines.join('\n'), 'utf8');
    console.log(`Patched successfully from line ${startIndex} to ${endIndex}`);
} else {
    console.log('Failed to find boundaries');
}
